### PR TITLE
Add GitHub action to build and push a container to hub.docker.com

### PR DESCRIPTION
**Pull request by craighurley**
This PR adds a GitHub action to build and push a container to hub.docker.com in response to a tag being pushed to this repo.

# Initial Setup

## On hub.docker.com

- Create an account if you don't have one already, e.g. `username`.
- Create a token.
- Create a repo, e.g. `org-formation-cli`

## On GitHub

create the following secrets in this repo, `org-formation/org-formation-cli`:

- `DOCKER_USERNAME`, your username for hub.docker.com, e.g. `username`.
- `DOCKER_TOKEN`, the token associated with the username on hub.docker.com.
- `DOCKER_REPOSITORY`, the name of the user and repo on hub.docker.com, e.g. `username/org-formation-cli`

# Description

Once that's done, when you push a tag/RELEASE_VERSION to the repo, the GitHub action will do the following:

1. Login to hub.docker.com
2. Build a new container, passing the pushed tag/RELEASE_VERSION to the docker build, which is then used as the RELEASE_VERSION in the command: `npm install -g --production aws-organization-formation@${RELEASE_VERSION}`.
3. Push the container to hub.docker.com with the following tags:

- RELEASE_VERSION, e.g. `0.9.12`.
- `latest`, where `latest` will always have the same digest as the most recently pushed tag.

# Potential Issue

I marked this PR as draft as one potential issue I see with this is if your existing build process pushes to npm _after_ you tag this repo.  If you push to npm before you tag this repo, then it's o.k.  Let me know what your process is.

### CODE REVIEW

**Comment by craighurley** [comment]
Also, I need to update the `README.md` with your hub.docker.com username after you creat it.

**Comment by eduardomourar** [comment]
@craighurley , we have created a new organization in Docker Hub and all secrets have been setup for this repository. Here is the link to the docker repository: https://hub.docker.com/repository/docker/orgformation/org-formation-cli

File: README.md [hunk 1]
```diff
@@ -41,6 +41,23 @@ You can now execute the command line program `org-formation`. try:
...
 
+### Docker
+
+If you choose, you can run org-formation in a docker container:
+
+```sh
+# Set the AWS_PROFILE environment variable and pass it to the container
+> AWS_PROFILE=example
+# Run the container
+> docker run --rm -it -v $HOME/.aws:/root/.aws:ro -v $PWD:/workdir -w /workdir -e AWS_PROFILE <TODO:CONFIRM_DOCKER_HUB_USER>/org-formation-cli
```
Comment (eduardomourar) [inline README.md:52]: we can the right repository here: `orgformation/org-formation-cli`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# release version passed in on build, just default it temporarily
+ARG         RELEASE_VERSION=0.0.1
+
+FROM        node:13.10-alpine3.11
```
Comment (eduardomourar) [inline Dockerfile:4]: i would prefer to use the latest LTS version, so `node:12.18-alpine3.11`

File: README.md (hunk 1 above)
Comment (craighurley) [reply README.md]: Updated with official name from hub.docker.com.

File: Dockerfile (hunk 2 above)
Comment (craighurley) [reply Dockerfile]: Changed to LTS version

**Review by eduardomourar — APPROVED** [review]
@OlafConijn , LGTM

**Review by OlafConijn — APPROVED** [review]
i think this is brilliant. thanks @craighurley!

**Pull request merged**